### PR TITLE
Payment Channel CLI tests

### PR DESCRIPTION
**Pull request by dirkmc**
This approach
- moves some common code in the integration tests to a shared place and calls it from the payment channel test
- creates a mock CLI App, and uses it to run individual CLI commands
- exposes API servers for each instance and clients to access them

The test
- sets up a miner and two full nodes
- starts the miner mining blocks
- sets up accounts for a creator and receiver
- calls various CLI commands to set up a channel, create vouchers, add vouchers, settle and collect

### CODE REVIEW

File: api/test/paych.go [hunk 1]
```diff
@@ -249,73 +248,3 @@ func waitForMessage(ctx context.Context, t *testing.T, paymentCreator TestNode,
 	fmt.Println("Confirmed", desc)
 	return res
 }
-
```
Comment (dirkmc) [inline api/test/paych.go:252]: Move this to common code

File: api/test/blockminer.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package test
```
Comment (dirkmc) [inline api/test/blockminer.go:1]: Extracted this from paych test

File: api/test/util.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package test
```
Comment (dirkmc) [inline api/test/util.go:1]: Extracted from paych test

File: node/node_test.go [hunk 4]
```diff
@@ -62,435 +28,12 @@ func init() {
 	verifreg.MinVerifiedDealSize = big.NewInt(256)
 }
 
-func testStorageNode(ctx context.Context, t *testing.T, waddr address.Address, act address.Address, pk crypto.PrivKey, tnd test.TestNode, mn mocknet.Mocknet, opts node.Option) test.TestStorageNode {
```
Comment (dirkmc) [inline node/node_test.go:65]: Extracted a bunch of code to a common package

File: node/test/builder.go [hunk 5]
```diff
@@ -0,0 +1,504 @@
+package test
```
Comment (dirkmc) [inline node/test/builder.go:1]: Extracted this code from node integration tests

**Comment by dirkmc** [comment]
Yeah the nice thing about this approach is you have access to the full API and lotus internals, so for example when we settle, we can get the settle height and then use the chain events API to wait until that height before calling collect

File: api/test/paych.go [hunk 6]
```diff
@@ -249,73 +248,3 @@ func waitForMessage(ctx context.Context, t *testing.T, paymentCreator TestNode,
...
-	}()
-}
-
-func (bm *blockMiner) stop() {
-	atomic.AddInt64(&bm.mine, -1)
-	fmt.Println("shutting down mining")
-	<-bm.done
-}
-
-func sendFunds(ctx context.Context, t *testing.T, sender TestNode, addr address.Address, amount abi.TokenAmount) {
```
Comment (dirkmc) [inline api/test/paych.go:297]: Extracted sendFunds to common code (util.go)

**Review by magik6k — APPROVED** [review]

**Pull request merged**